src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -141,7 +142,10 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
     if (this.state.showError) {
       return <DataSourceErrorMenu application={this.props.application} />;
     }
-    const label = this.state.selectedOption.length > 0 ? this.state.selectedOption[0].label : '';
+    const label =
```

Comment:
nit: can be cleaned up with: 
```
const label = this.state.selectedOption[0]?.label ?? '';
```
as TypeScript returns undefined when array is empty

Reply:
Got it. Thank you Rocky!